dashboard/entry/tagcheck.go

```diff
@@ -0,0 +1,43 @@
+package entry
+
+import (
+	"fmt"
+
+	"github.com/jinzhu/gorm"
+	"github.com/traggo/server/generated/gqlmodel"
+	"github.com/traggo/server/model"
+)
+
+func tagsDuplicates(src []*gqlmodel.InputTimeSpanTag, dst []*gqlmodel.InputTimeSpanTag) *gqlmodel.InputTimeSpanTag {
+	existingTags := make(map[string]struct{})
+	for _, tag := range src {
+		existingTags[tag.Key+":"+tag.Value] = struct{}{}
+	}
+
+	for _, tag := range dst {
+		if _, ok := existingTags[tag.Key+":"+tag.Value]; ok {
+			return tag
+		}
+
+		existingTags[tag.Key] = struct{}{}
+	}
```

Comment:
Can be simplified to:
```suggestion
func tagsDuplicates(src []*gqlmodel.InputTimeSpanTag, dst []*gqlmodel.InputTimeSpanTag) *gqlmodel.InputTimeSpanTag {
	existingTags := make(map[gqlmodel.InputTimeSpanTag]struct{})
	for _, tag := range src {
		existingTags[*tag] = struct{}{}
	}

	for _, tag := range dst {
		if _, ok := existingTags[*tag]; ok {
			return tag
		}
	}

	return nil
}
```